ui/src/warp_runner/mod.rs

```diff
@@ -110,110 +123,181 @@ async fn handle_login(notify: Arc<Notify>) {
+                        continue;
+                    }
+                    _ => {
+                        log::error!("multipass.get_own_identity failed: {}", e);
+                        return Err(e);
+                    }
+                },
+            }
         }
+    };
+
+    tokio::select! {
+        r = multipass_init_done => r,
+        _ = notify.notified() => {
+            log::error!("notified interrupted multipass initialization");
+            Err(Error::Other)
+        },
     }
 }
 
-// tesseract needs to be initialized before warp is initialized. this function does just that
-async fn login(
-    passphrase: &str,
-    tesseract: Tesseract,
-    new_account: bool,
-) -> Result<manager::Warp, warp::error::Error> {
-    log::debug!("login");
-    tesseract.unlock(passphrase.as_bytes())?;
-    while !tesseract.is_unlock() {
-        tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-    }
+// don't set file or autosave until tesseract is unlocked
+// assumes that all anyone needs from tesseract is "keypair"
+// otherwise, Tesseract::to_file probably needs to call file.sync_all()
+async fn init_tesseract(overwrite_old_account: bool) -> Result<Tesseract, Error> {
+    log::trace!("initializing tesseract");
+
+    let configure_tesseract = |tesseract: Tesseract| {
+        // prevent other things from corrupting the real tesseract file.
+        tesseract.set_file(STATIC_ARGS.warp_path.join("fake_tesseract.json"));
+        tesseract.set_autosave();
+        tesseract
+    };
+
+    // this code path addresses cross-platform issues involving account recreation.
+    // the tesseract file was being overwritten incorrectly.
+    // to fix this, manually delete the file and re-create it.
+    if overwrite_old_account {
+        // delete old account data
+        if let Err(e) = std::fs::remove_dir_all(&STATIC_ARGS.uplink_path) {
+            log::warn!("failed to delete uplink directory: {}", e);
+        }
+
+        // create directories
+        if let Err(e) = std::fs::create_dir_all(&STATIC_ARGS.warp_path) {
+            log::warn!("failed to create warp directory: {}", e);
+        }
+
+        // create the tesseract key file so it can be saved later
+        if let Err(e) = std::fs::File::create(&STATIC_ARGS.tesseract_path) {
+            log::error!("failed to create tesseract file: {}", e);
+            return Err(warp::error::Error::CannotSaveTesseract);
+        }
 
-    if !new_account && !tesseract.exist("keypair") {
-        log::info!("string keypair not found in tesseract");
-        return Err(warp::error::Error::IdentityNotCreated);
+        return Ok(configure_tesseract(Tesseract::default()));
     }
-    let res = warp_initialization(tesseract, false).await;
-    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-    res
+
+    // open existing file or create new one
+    let tesseract = match std::fs::File::open(&STATIC_ARGS.tesseract_path) {
```

Comment:
This could be replaced with `Tesseract::open_or_create`, which would open the file if it exist, otherwise it would attempt to create one if it doesnt. Do note, it would only attempts this if the file doesnt exist. If it does exist, it may throw other errors ( (eg doesnt have permissions, OOM, etc), in which case those errors should be brought up stream